tests/filecheck/backend/riscv/riscv_arith_lowering_rv32.mlir

```diff
@@ -0,0 +1,145 @@
+// RUN: xdsl-opt -p riscv-lower-arith-rv32 %s | xdsl-opt --print-op-generic | filecheck %s
+"builtin.module"() ({
+    %lhsi32 = "arith.constant"() {value = 1 : i32} : () -> i32
+    // CHECK: %{{.*}} = "riscv.li"() {"immediate" = 1 : si32} : () -> !riscv.reg<>
+    %rhsi32 = "arith.constant"() {value = 2 : i32} : () -> i32
+    // CHECK-NEXT: %{{.*}} = "riscv.li"() {"immediate" = 2 : si32} : () -> !riscv.reg<>
+    %lhsindex = "arith.constant"() {value = 1 : index} : () -> index
+    // CHECK-NEXT: %{{.*}} = "riscv.li"() {"immediate" = 1 : si32} : () -> !riscv.reg<>
+    %rhsindex = "arith.constant"() {value = 2 : index} : () -> index
+    // CHECK-NEXT: %{{.*}} = "riscv.li"() {"immediate" = 2 : si32} : () -> !riscv.reg<>
+    %lhsf32 = "arith.constant"() {value = 1.000000e+00 : f32} : () -> f32
+    // CHECK-NEXT: %{{.*}} = "riscv.li"() {"immediate" = 1065353216 : si32} : () -> !riscv.reg<>
```

Comment:
Sorry if I missed this before, but why are we using signed types here?
Seems like a no-no since LLVM does not use it?

Reply:
That's because the `LiOp` constructor does the following : 
```python 
immediate = IntegerAttr(immediate, IntegerType(32, Signedness.SIGNED))
```

Reply:
I will make an issue for this!

Reply:
#1216 was made!